wolfcrypt/test/test.c

```diff
@@ -25012,10 +25012,73 @@ static int crypto_ecc_sign(const uint8_t *key, uint32_t keySz,
 
     return ret;
 }
+#endif /* HAVE_ECC_SIGN && HAVE_ECC_VERIFY */
+
+#ifdef HAVE_ECC_DHE
+/* 
+ * Currently we only have support for P-256/384 for non-blocking key gen and
+ * shared secret gen.
+ */
+static int ecc_test_nonblock_dhe(WC_RNG* rng) {
+    int ret = 0;
+    ecc_key keyA;
+    ecc_key keyB;
+    ecc_nb_ctx_t nbCtxA;
+    ecc_nb_ctx_t nbCtxB;
+    byte secretA[ECC_SHARED_SIZE];
+    byte secretB[ECC_SHARED_SIZE];
+    word32 secretSzA = ECC_SHARED_SIZE;
+    word32 secretSzB = ECC_SHARED_SIZE;
+
+    if (ret == 0) {
+        ret = wc_ecc_init(&keyA);
+    }
+    if (ret == 0) {
+        ret = wc_ecc_init(&keyB);
+    }
+    if (ret == 0) {
+        ret = wc_ecc_set_nonblock(&keyA, &nbCtxA);
+    }
+    if (ret == 0) {
+        ret = wc_ecc_set_nonblock(&keyB, &nbCtxB);
+    }
+    if (ret == 0) {
+        do {
+            ret = wc_ecc_make_key_ex(rng, ECC_CURVE_SZ, &keyA, ECC_CURVE_ID);
+        } while (ret == FP_WOULDBLOCK);
```

Comment:
Please add `count` and print like we do on the other examples. The number of iterations to complete each is useful data.
```
#ifdef DEBUG_WOLFSSL
    printf("ECC non-block key gen: %d times\n", count);
#endif
```